hazelcast/src/main/java/com/hazelcast/cache/impl/CachePartitionSegment.java

```diff
@@ -91,11 +104,28 @@ public void destroy() {
         clear();
     }
 
-    public boolean hasAnyRecordStore() {
-        return !recordStores.isEmpty();
+    public void close() {
+        synchronized (mutex) {
+            for (ICacheRecordStore store : recordStores.values()) {
+                store.close();
+            }
+        }
+        recordStores.clear();
     }
 
-    public boolean hasRecordStore(String name) {
-        return recordStores.containsKey(name);
+    protected static void transferRecordData(ICacheRecordStore src, ICacheRecordStore dest) {
```

Comment:
ok, but at the moment it is not a prototype solution but a big feature/module. So, I prefer moving this responsibility to `ICacheRecordStore` anymore.